### PR TITLE
`Programming exercises`: Move the open repository buttons to the new code popup

### PR DESCRIPTION
**Pull request by pzdr7**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

#### Changes affecting Programming Exercises
-  [x] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).

### Motivation and Context

- We want to have a more consistent experience when it comes to opening the repository in a readonly view
- We want to reduce the number of buttons displayed to students when they work on programming exercises.

### Description

- Renamed the "Clone repository" button to "Code". In the future, we may move the online code editor / online IDE buttons into this popup as well (inspired by the GitHub "Code" button).
- Added a link (for LocalVC; others are unchanged) to the Code popup that opens the repository view

### Steps for Testing

Prerequisites:
- 1 Instructor
- Test Server with LocalVC enabled
- 1 Programming exercise with 1 participation, manual assessment enabled, and both online editor and local IDE enabled

**Student view** (for easier testing, do this step as an instructor. you will need instructor access for the steps later)

1. Log in
2. Go to Courses > Your course > Your programming exercise
3. Start your participation (if you haven't yet)
4. Check that there is no "Open Repository" button
5. Click on the "Code" button
6. Verify that you can see the image button and click it.
7. It should take you to a new tab with the repository view.

**Instructor: Exercise participations**:

8. Go back to the exercise
9. Go to Instructor actions: Participations
10. Verify that there is one column with the "Code" button
11. Check that the button that links to the repository works as expected

**Instructor: Exercise scores**

12. Go back to the exercise
13. Go to Instructor actions: Scores
14. Check that the Code button works as for the previous steps

**Instructor: Template/Solution/Tests repos**:

15. Go back to the exercise
16. Go to instructor actions: View
17. For each of the template/solution/tests repos, check if the Code button works as for the previous steps. Note that we currently do not support viewing the Auxiliary Repositories.

**Instructor: Manual assessment example solution**

18. Go to the Assessment Dashboard (important: do not continue past the initial page. if you do not see the "I have read and understood the instructions" button, use a different account)
19. Scroll down to image
20. Check that the "Code" button works as expected.

#### Exam Mode Testing

Prerequisites:
- 1 Testserver with LocalVC enabled
- 1 Exam with a programming exercise (enable both the online editor and the local IDE)
- 1 Student
- 1 Instructor

**Student participation and summary**

1. Participate in the exam as a student. Make a submission using the online editor.
2. Check that you can use the "Code" button during the exam to open the repository in a new tab.
3. Hand in the exam and view the summary. Check that you can use the Code button there to open the repository this way.

**Instructor: summary**

4. As the instructor, go to course management > your course > your exam > student exams
5. View the student exam you submitted and go to its summary.
6. Verify that the Code button works as expected

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [x] Test 1
- [x] Test 2

### Test Coverage

Only non-trivial changes in .ts files; the rest of the changes are in either the playwright tests or HTML templates

#### Student view (regular)

<details>
  <summary>Before</summary>

image
</details>

<details>
  <summary>After</summary>

image

The repository view can now be accessed via the button in the popup:

image

</details>

#### Instructor: Exercise participations

<details>
  <summary>Before</summary>

image

</details>

<details>
  <summary>After</summary>

image

</details>

#### Instructor: Exercise scores

<details>
  <summary>Before</summary>

image
</details>

<details>
  <summary>After</summary>

image

</details>

#### Tutor/Instructor: Example solution (manual assessment)

<details>
  <summary>Before</summary>

image
</details>

<details>
  <summary>After</summary>

image
</details>

#### Exam participation / summary

<details>
  <summary>Before</summary>

Participation
image

Summary
image
</details>

<details>
  <summary>After</summary>

Participation
image

Summary
image

</details>

## Summary by CodeRabbit

- **New Features**
  - Updated button labels and conditional rendering for SSH and local version control settings in code buttons.
  - Enhanced visibility and interaction options for repository management in both student and instructor views.

- **Documentation**
  - Refined exam conduction, exercise saving, and repository access instructions in user guides.

- **Bug Fixes**
  - Simplified and clarified repository link retrieval methods for better user experience.

- **Localization**
  - Removed outdated translations related to repository access in German and English localization files.

### CODE REVIEW

File: src/main/webapp/app/shared/components/code-button/code-button.component.html [hunk 1]
```diff
@@ -39,28 +39,40 @@ <h5>{{ cloneHeadline | artemisTranslate }}</h5>
...
-                    <a
-                        style="flex: 1"
-                        class="btn btn-secondary btn-sm me-2 open-repository-button"
-                        [href]="getHttpRepositoryUri() | safeUrl"
-                        target="_blank"
-                        rel="noopener noreferrer"
-                    >
-                        <fa-icon [icon]="faExternalLink" />
-                    </a>
+                    @if (!localVCEnabled) {
```
Comment (pzdr7) [inline src/main/webapp/app/shared/components/code-button/code-button.component.html:55]: I'm open to suggestions that reduce the redundancy here - unfortunately, I cannot attach a `href` and `routerLink` to the same anchor, as even `[routerLink]="undefined"` will overwrite the `href`

**Review by JohannesStoehr — APPROVED** [review]
Code changes look good to me.
One thing (for a follow up): With Bitbuket it was possible to simply take the clone url and past it into the browser to see the repository. Maybe we could also offer this for LocalVC

**Review by BBesrour — APPROVED** [review]
Tested on TS3, works as described!
Also for exam mode.

Isn't the button to open the repository a bit too hidden though? it took me a few seconds to find while I was looking for it.

**Review by egekurt123 — DISMISSED** [review]
Code LGTM

**Review by coolchock — DISMISSED** [review]
tested on ts4, works as expected!

**Review by JohannesStoehr — DISMISSED** [review]
Reapprove code

**Review by BBesrour — DISMISSED** [review]
re-approve

**Review by coolchock — DISMISSED** [review]
re-approve

**Review by egekurt123 — DISMISSED** [review]
Reapprove Code LGTM

File: docs/user/exercises/programming-integrated-code-lifecycle-repository-view.inc [hunk 2]
```diff
@@ -7,25 +7,26 @@ For Students
...
+To the right side of the screen, above the problem statement, you can see the result of the submission. You can also copy the repository URI by pressing
+the '*Code*' button to clone the repository to your local machine and see the full commit history by pressing the '*Open Commit History*' button. Finally, you can
+download the repository as a ZIP file by pressing the '*Download Repository*' button.
 
 .. figure:: programming/current-repository-content-view.png
    :alt: Current Repository Content
    :align: center
 
-   Current Repository Content
+   Current Repository C ontent
```
Comment (pzdr7) [inline docs/user/exercises/programming-integrated-code-lifecycle-repository-view.inc:29]:
```suggestion
   Current Repository Content
```

**Review by az108 — COMMENTED** [review]
The only trouble i faced was trying to test sumbit my code as this error came up on the test server. But this is unrelated to the PR as far as I am aware. 
image

**Review by az108 — APPROVED** [review]
Tested on TS1 all Steps for a normal programming Exercise and then created an exam with a programming exercise and tested it aswell. Everything works as expected and I like the changes 😄 . Code LGTM aswell 👍

**Review by JohannesStoehr — APPROVED** [review]
Reapprove code. Also looked at the documentation changes

**Review by BBesrour — APPROVED** [review]
re-approve, docs lgtm

**Review by krusche — APPROVED** [review]
Tested the changes, works well 👍

**Pull request merged**